Achievements.lua

```diff
@@ -0,0 +1,146 @@
+-- Globale Tabelle für alle Achievements
+SchlingelInc.Achievements = {}
+
+-- Tabelle für gespeicherte Erfolge
+SchlingelInc.Achievements.completedAchievements = {}
+
+function SchlingelInc.Achievements:AnnounceToGuild(achievement)
+    local name = SchlingelInc:RemoveRealmFromName(UnitName("player"))
+    local msg = string.format("%s hat den Erfolg '%s' erspielt!", name, achievement)
+    print(msg) --debug
+    -- SendChatMessage(msg, "GUILD")
+end
+
+-- Funktion, um Erfolge zu prüfen
+function SchlingelInc.Achievements:CheckAchievements()
+    for _, achievement in ipairs(self.achievements) do
+        -- Prüfen, ob der Erfolg bereits abgeschlossen ist
+        if not self.completedAchievements[achievement.name] and achievement.condition() then
+            -- Erfolg als abgeschlossen markieren
+            self.completedAchievements[achievement.name] = true
+            SavedAchievements[achievement.name] = true                                            -- Direkt in SavedAchievements speichern
+            self.savedAchievementPoints = (self.savedAchievementPoints or 0) + achievement.points -- Punkte hinzufügen
```

Comment:
Why do you need to default `savedAchievementPoints` here? Isn't it certain, that it is already initilaized? 👀